src/Mod/CAM/Path/Machine/models/machine.py

```diff
@@ -0,0 +1,1474 @@
+            },
+            "version": self.version,
+        }
+
+        # Add post-processor configuration
+        data["postprocessor"] = {
+            "file_name": self.postprocessor_file_name,
+            "args": self.postprocessor_args,
+            "motion_mode": self.motion_mode.value,
+        }
+
+        # Output options
+        data["output"] = {
+            "comments": self.output.comments,
+            "blank_lines": self.output.blank_lines,
+            "header": self.output.header,
+            "line_numbers": self.output.line_numbers,
+            "path_labels": self.output.path_labels,
+            "machine_name": self.output.machine_name,
+            "doubles": self.output.doubles,
+            "output_units": self.output.output_units.value,
+            "axis_precision": self.output.axis_precision,
+            "feed_precision": self.output.feed_precision,
+            "spindle_decimals": self.output.spindle_decimals,
+            "command_space": self.output.command_space,
+            "comment_symbol": self.output.comment_symbol,
+            "line_increment": self.output.line_increment,
+            "line_number_start": self.output.line_number_start,
+            "end_of_line_chars": self.output.end_of_line_chars,
+        }
+
+        # G-code blocks (only non-empty ones)
+        blocks = {}
+        if self.blocks.pre_job:
+            blocks["pre_job"] = self.blocks.pre_job
+        if self.blocks.post_job:
+            blocks["post_job"] = self.blocks.post_job
+        if self.blocks.preamble:
+            blocks["preamble"] = self.blocks.preamble
+        if self.blocks.postamble:
+            blocks["postamble"] = self.blocks.postamble
+        if self.blocks.safetyblock:
+            blocks["safetyblock"] = self.blocks.safetyblock
+        if self.blocks.pre_operation:
+            blocks["pre_operation"] = self.blocks.pre_operation
+        if self.blocks.post_operation:
+            blocks["post_operation"] = self.blocks.post_operation
+        if self.blocks.pre_tool_change:
+            blocks["pre_tool_change"] = self.blocks.pre_tool_change
+        if self.blocks.post_tool_change:
+            blocks["post_tool_change"] = self.blocks.post_tool_change
+        if self.blocks.tool_return:
+            blocks["tool_return"] = self.blocks.tool_return
+        if self.blocks.pre_fixture_change:
+            blocks["pre_fixture_change"] = self.blocks.pre_fixture_change
+        if self.blocks.post_fixture_change:
+            blocks["post_fixture_change"] = self.blocks.post_fixture_change
+        if self.blocks.pre_rotary_move:
+            blocks["pre_rotary_move"] = self.blocks.pre_rotary_move
+        if self.blocks.post_rotary_move:
+            blocks["post_rotary_move"] = self.blocks.post_rotary_move
+
+        if blocks:
+            data["blocks"] = blocks
+
+        # Processing options
+        data["processing"] = {
+            "modal": self.processing.modal,
```

Comment:
This needs research.  I'm not sure if it's handled by any postprocessor now.   We need to understand the use-cases better.